website/guides/advanced/schemas.md.erb

```diff
@@ -0,0 +1,325 @@
+* You need to concatenate `first_name` and `last_name` into a `name` field. (Suppose they didn't read
+  [Falsehoods about names](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/)).
+
+
+### Example: Mooshing together name fields
+
+Let's pretend one of your teammates falsely assumed folks always have first and last names, so we have a `first_name`
+and a `last_name` field coming from a source, and we'd like to output a `name` field to a sink.
+
+```toml title="config.toml"
+[transforms.moosh_names]
+  type = "add_fields"
+  inputs = ["source0"]
+  [transform.moosh_names.fields]
+    field = "name"
+    value = "{{first_name}} {{last_name}}"
+
+[transforms.drop_old_names]
+  type = "remove_fields"
+  inputs = ["moosh_names"]
+  fields = ["first_name", "last_name"]
+```
+
+What if you had to do this in reverse? Try using the [`regex`][transform_regex_parser] or
+[`split`][transform_split] transforms.
+
+
+## Coercing Data Types
+
+Occasionally services will provide you with data that is in the right shape, but the types are wrong. Perhaps a string
+should be a number, or vice versa.
+
+The [`coercer`][transform_coercer] transform is the correct tool for this job!
+
+```toml title="config.toml"
+[transforms.correct_source_types]
+  type = "coercer"
+  inputs = ["source0"]
+  types.count = "int"
+  types.date = "timestamp|%F"
+```
+
+### Example: Coercing between date formats
+
+There are a lot of ways to represent time. In the US folks tend to use `MM/DD/YYYY` or the (more reasonable)
+`YYYY/MM/DD` which Canada and China like. In the EU, South America, and Africa they prefer `DD/MM/YYYY`. Like personal
+identities, all are valid. Vector lets us take in timestamps and output specific formats easily.
+
+To do this we'll use `timestamp| $FORMAT`. To build a `$FORMAT`, we can reference the
+[`strftime`](https://docs.rs/chrono/0.4.10/chrono/format/strftime/index.html) documentation. Let's ship some Canadian
+friendly logs up to the great white north!
+
+```toml title="config.toml"
+[transforms.format_timestamp]
+  type = "coercer"
+  types.timestamp = "timestamp|%Y/%m/%d:%H:%M:%S %z"
+```
+
+## Working with data formats
+
+Not all logs come structured. Some services provide JSON, some provide plaintext, others ship around protobufs. With
+Vector you can handle them all.
+
+Generally Vector will be able to determine the encodings to use by the source or sink used. In some cases, multiple are
+supported. In these cases, you can use the `encoding` option.
+
+The [`console`](/docs/reference/sinks/console/#encoding) sink supports both `json` and `text` as its
+output format
+
+```toml title="config.toml"
+[sinks.print]
+  type = "console"
+  inputs = ["source0"]
+  target = "stdout"
+  encoding = "json"
+```
+
+You can also use a transform like [`json_parser`][transform_json_parser] or
+[`grok_parser`][transform_grok_parser] to parse out data in a given field.
+
+## Using a Schema
```

Comment:
I like this section, but I think we should also make schema management progressive. The way I think about it is (sorted by simplest first)

1. Define a static schema with the `coercer` transform and set `drop_unspecified` to `true`.
2. Use the `json_schema` transform (pending)
3. Use the `aws_glue_schema` transform (pending)
4. Extend Vector with WASM to use protobufs, etc. (pending)

Given that 2-4 are pending I think we can stop with 1 and complete this guide. In fact, I'd probably move 3 and 4 into their own guides since they are much more advanced.